checks/libs/wmi/sampler.py

```diff
@@ -250,33 +253,84 @@ def _get_connection(self):
 
         return connection
 
+    """
+    Builds filter from a filter list.
+    - filters: expects a list of dicts, typically:
+            - [{'Property': value},...] or
+            - [{'Property': (comparison_op, value)},...]
+
+            NOTE: If we just provide a value we defailt to '=' comparison operator.
+            Otherwise, specify the operator in a tuple as above: (comp_op, value)
+
+            The filters also support lists/tuples as property 'values' to filter.
+            When such a list is provided, we OR filter for all such property values:
+            - [{'Property': [value1, value2, value3]},...] or
+            - [{'Property': [(cmp_op, value1), (cmp_op, value2)]},...] or
+
+    Normally the filter is expected to be exclusive, such that all filter conditions
+    should be met - therefore we AND all filters. If you'd like to OR, then set
+    inclusive to True.
+    """
     @staticmethod
-    def _format_filter(filters):
+    def _format_filter(filters, inclusive=False):
         """
         Transform filters to a comprehensive WQL `WHERE` clause.
         """
-        def build_where_clause(fltr):
+        def build_where_clause(fltr, inclusive):
```

Comment:
You don't change the value of `inclusive` across your recursive calls, so I guess you don't need to have this argument redefined in `build_where_clause ` but only in `_format_filter`.